src/Aspire.Hosting.RabbitMQ/RabbitMQBuilderExtensions.cs

```diff
@@ -58,5 +60,21 @@ public static IResourceBuilder<RabbitMQServerResource> WithDataVolume(this IReso
     /// <param name="isReadOnly">A flag that indicates if this is a read-only mount.</param>
     /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
     public static IResourceBuilder<RabbitMQServerResource> WithDataBindMount(this IResourceBuilder<RabbitMQServerResource> builder, string source, bool isReadOnly = false)
-        => builder.WithBindMount(source, "/var/lib/rabbitmq", isReadOnly);
+        => builder.WithBindMount(source, "/var/lib/rabbitmq", isReadOnly)
+                  .RunWithStableNodeName();
+
+    private static IResourceBuilder<RabbitMQServerResource> RunWithStableNodeName(this IResourceBuilder<RabbitMQServerResource> builder)
+    {
+        if (builder.ApplicationBuilder.ExecutionContext.IsRunMode)
```

Comment:
What happens when you publish?

Reply:
The assumption is when you publish the host name of the container is stable, rather than what happens in dev where we randomize the port every time.

Reply:
More details about how RabbitMQ auto-assigns node names in their docs: https://www.rabbitmq.com/docs/clustering#node-names and https://www.rabbitmq.com/docs/configure

From the latter link:

> The node name should be unique per Erlang-node-and-machine combination. To run multiple nodes, see the [clustering guide](https://www.rabbitmq.com/docs/clustering).

> Default:
>   Unix: rabbit@$HOSTNAME
>   Windows: rabbit@%COMPUTERNAME%

So ultimately, what happens after publish depends on what `$HOSTNAME` results in on the container where RabbitMQ is running.